src/modules/airspeed_selector/airspeed_selector_main.cpp

```diff
@@ -617,119 +625,159 @@ void AirspeedModule::select_airspeed_and_publish()
 			 */
-			events::send<uint8_t, uint8_t>(events::ID("airspeed_selector_estimation_regain"), events::Log::Info,
-						       "Airspeed sensor healthy, start using again", _prev_airspeed_index,
-						       _valid_airspeed_index);
+			events::send<uint8_t, uint8_t>(events::ID("airspeed_selector_estimation_regain"), events::Log::Critical,
+						       "Airspeed sensor healthy, start using again", prev_airspeed_index,
+						       valid_airspeed_index);
 		}
 	}
 
-	_prev_airspeed_index = _valid_airspeed_index;
 	_prev_number_of_airspeed_sensors = _number_of_airspeed_sensors;
 
 	airspeed_validated_s airspeed_validated = {};
 	airspeed_validated.timestamp = _time_now_usec;
-	airspeed_validated.true_ground_minus_wind_m_s = NAN;
 	airspeed_validated.calibrated_ground_minus_wind_m_s = NAN;
+	airspeed_validated.calibraded_airspeed_synth_m_s = NAN;
 	airspeed_validated.indicated_airspeed_m_s = NAN;
 	airspeed_validated.calibrated_airspeed_m_s = NAN;
 	airspeed_validated.true_airspeed_m_s = NAN;
-	airspeed_validated.airspeed_sensor_measurement_valid = false;
-	airspeed_validated.selected_airspeed_index = _valid_airspeed_index;
 
-	airspeed_validated.airspeed_derivative_filtered = _airspeed_validator[_valid_airspeed_index -
-					      1].get_airspeed_derivative();
-	airspeed_validated.throttle_filtered = _airspeed_validator[_valid_airspeed_index - 1].get_throttle_filtered();
-	airspeed_validated.pitch_filtered = _airspeed_validator[_valid_airspeed_index - 1].get_pitch_filtered();
+	airspeed_validated.airspeed_derivative_filtered = _airspeed_validator[valid_airspeed_index -
+					     1].get_airspeed_derivative();
+	airspeed_validated.throttle_filtered = _airspeed_validator[valid_airspeed_index - 1].get_throttle_filtered();
+	airspeed_validated.pitch_filtered = _airspeed_validator[valid_airspeed_index - 1].get_pitch_filtered();
 
-	switch (_valid_airspeed_index) {
-	case airspeed_index::DISABLED_INDEX:
+	airspeed_validated.airspeed_source = valid_airspeed_index;
+	_prev_airspeed_src = _valid_airspeed_src;
+
+	switch (_valid_airspeed_src) {
+	case AirspeedSource::DISABLED:
 		break;
 
-	case airspeed_index::GROUND_MINUS_WIND_INDEX:
+	case AirspeedSource::GROUND_MINUS_WIND:
 		airspeed_validated.indicated_airspeed_m_s = _ground_minus_wind_CAS;
 		airspeed_validated.calibrated_airspeed_m_s = _ground_minus_wind_CAS;
 		airspeed_validated.true_airspeed_m_s = _ground_minus_wind_TAS;
 		airspeed_validated.calibrated_ground_minus_wind_m_s = _ground_minus_wind_CAS;
-		airspeed_validated.true_ground_minus_wind_m_s = _ground_minus_wind_TAS;
 
 		break;
 
+	case AirspeedSource::SYNTHETIC: {
+			airspeed_validated.throttle_filtered = _airspeed_validator[0].get_throttle_filtered();
+			airspeed_validated.pitch_filtered = _airspeed_validator[0].get_pitch_filtered();
+
+			float synthetic_airspeed;
+
+			if (airspeed_validated.throttle_filtered < _param_fw_thr_trim.get() && _param_fw_thr_aspd_min.get() > 0.f) {
```

Comment:
With the current assumption of direct link between throttle and airspeed we ofc neglect the pitch angle dependency on airspeed. 
Maybe we can find a clean mapping using the max/min pitch angles and max/min climb/descend rates.